### ISSUE TITLE
regplot: IndexError: too many indices for array

### ISSUE THREAD

**Issue by banyh** [mwaskom/seaborn#2012]
```
import seaborn as sns
sns.regplot(x=[1], y=[1])
```
will cause following error:
```
IndexError                                Traceback (most recent call last)
<ipython-input-3-4dd949c43459> in <module>
----> 1 sns.regplot(x=[1], y=[1])

~/venv/lib/python3.7/site-packages/seaborn/regression.py in regplot(x, y, data, x_estimator, x_bins, x_ci, scatter, fit_reg, ci, n_boot, units, seed, order, logistic, lowess, robust, logx, x_partial, y_partial, truncate, dropna, x_jitter, y_jitter, label, color, marker, scatter_kws, line_kws, ax)
    808                                  order, logistic, lowess, robust, logx,
    809                                  x_partial, y_partial, truncate, dropna,
--> 810                                  x_jitter, y_jitter, color, label)
    811 
    812     if ax is None:

~/venv/lib/python3.7/site-packages/seaborn/regression.py in __init__(self, x, y, data, x_estimator, x_bins, x_ci, scatter, fit_reg, ci, n_boot, units, seed, order, logistic, lowess, robust, logx, x_partial, y_partial, truncate, dropna, x_jitter, y_jitter, color, label)
    112         # Drop null observations
    113         if dropna:
--> 114             self.dropna("x", "y", "units", "x_partial", "y_partial")
    115 
    116         # Regress nuisance variables out of the data

~/venv/lib/python3.7/site-packages/seaborn/regression.py in dropna(self, *vars)
     64             val = getattr(self, var)
     65             if val is not None:
---> 66                 setattr(self, var, val[not_na])
     67 
     68     def plot(self, ax):

IndexError: too many indices for array
```

**This bug only exists in 0.10.0, but not in 0.9.0.**

**Comment by MaozGelbart** [comment]
Hi @banyh , this has been reported in #1958 and is fixed on master (by #1969 ).

**Comment by mwaskom** [comment]
Thanks @MaozGelbart. Duplicate of #1949
